PR: https://github.com/odoo/odoo/pull/186906

From: 9144eb051538d5a92d403e1ad9c635ce39e090ec
From: Gorash
Date: 2024-11-14 10:39:02

Breaking data model changes scores: del:14 + add:0, change matches:
-    _inherit = ['account.bank.statement.line']
-    _inherit = ['account.journal']
-    _inherit = ['account.move']
-    _inherit = ['account.move.line']
-    _inherit = ['account.payment']
-    _inherit = ['barcode.rule']
-    _inherit = ['digest.digest']
-    _inherit = ['procurement.group']
-    _inherit = ['res.users']
-    _inherit = ['stock.picking']
-    _inherit = ['procurement.group']
-    _inherit = ['stock.move']
-    _inherit = ['stock.rule']
-    _inherit = ["stock.warehouse"]

Total Changes: 71

[REV] *: Restore the model `_name`

Following internal discussions, the python typing and inheritance task is
canceled. The use of the class name instead of '_name' was done to allow
notations such as `partner_id = fields.Many2one[base.ResParter](string='Partner')`.
Since this notation is canceled, we put `_name` back (although the class
name is sufficient).

Partial revert:
https://github.com/odoo/enterprise/pull/69762
https://github.com/odoo/odoo/pull/178200

closes odoo/odoo#186906

Related: odoo/enterprise#73643
Signed-off-by: Raphael Collet <rco@odoo.com>

================================= pseudo patch: =================================

--- a/addons/point_of_sale/models/account_bank_statement.py
+++ b/addons/point_of_sale/models/account_bank_statement.py
@@ -6,6 +6,6 @@ from odoo.exceptions import UserError
 
 
 class AccountBankStatementLine(models.Model):
-    _inherit = ['account.bank.statement.line']
+    _inherit = 'account.bank.statement.line'
 
     pos_session_id = fields.Many2one('pos.session', string="Session", copy=False)

--- a/addons/point_of_sale/models/account_fiscal_position.py
+++ b/addons/point_of_sale/models/account_fiscal_position.py
@@ -2,6 +2,7 @@ from odoo import models, api
 
 
 class AccountFiscalPosition(models.Model):
+    _name = 'account.fiscal.position'
     _inherit = ['account.fiscal.position', 'pos.load.mixin']
 
     @api.model

--- a/addons/point_of_sale/models/account_fiscal_position_tax.py
+++ b/addons/point_of_sale/models/account_fiscal_position_tax.py
@@ -2,6 +2,7 @@ from odoo import models, api
 
 
 class AccountFiscalPositionTax(models.Model):
+    _name = 'account.fiscal.position.tax'
     _inherit = ['account.fiscal.position.tax', 'pos.load.mixin']
 
     @api.model

--- a/addons/point_of_sale/models/account_journal.py
+++ b/addons/point_of_sale/models/account_journal.py
@@ -6,7 +6,7 @@ from odoo.exceptions import ValidationError
 
 
 class AccountJournal(models.Model):
-    _inherit = ['account.journal']
+    _inherit = 'account.journal'
 
     pos_payment_method_ids = fields.One2many('pos.payment.method', 'journal_id', string='Point of Sale Payment Methods')
 

--- a/addons/point_of_sale/models/account_move.py
+++ b/addons/point_of_sale/models/account_move.py
@@ -5,7 +5,7 @@ from odoo import fields, models, api
 
 
 class AccountMove(models.Model):
-    _inherit = ['account.move']
+    _inherit = 'account.move'
 
     pos_order_ids = fields.One2many('pos.order', 'account_move')
     pos_payment_ids = fields.One2many('pos.payment', 'account_move_id')
@@ -70,7 +70,7 @@ class AccountMove(models.Model):
 
 
 class AccountMoveLine(models.Model):
-    _inherit = ['account.move.line']
+    _inherit = 'account.move.line'
 
     def _stock_account_get_anglo_saxon_price_unit(self):
         self.ensure_one()

--- a/addons/point_of_sale/models/account_payment.py
+++ b/addons/point_of_sale/models/account_payment.py
@@ -5,7 +5,7 @@ from odoo import models, fields, api
 
 
 class AccountPayment(models.Model):
-    _inherit = ['account.payment']
+    _inherit = 'account.payment'
 
     pos_payment_method_id = fields.Many2one('pos.payment.method', "POS Payment Method")
     force_outstanding_account_id = fields.Many2one("account.account", "Forced Outstanding Account", check_company=True)

--- a/addons/point_of_sale/models/account_tax.py
+++ b/addons/point_of_sale/models/account_tax.py
@@ -6,6 +6,7 @@ from odoo.tools import split_every
 
 
 class AccountTax(models.Model):
+    _name = 'account.tax'
     _inherit = ['account.tax', 'pos.load.mixin']
 
     def write(self, vals):

--- a/addons/point_of_sale/models/account_tax_group.py
+++ b/addons/point_of_sale/models/account_tax_group.py
@@ -2,6 +2,7 @@ from odoo import api, models
 
 
 class AccountTaxGroup(models.Model):
+    _name = 'account.tax.group'
     _inherit = ['account.tax.group', 'pos.load.mixin']
 
     @api.model

--- a/addons/point_of_sale/models/barcode_rule.py
+++ b/addons/point_of_sale/models/barcode_rule.py
@@ -6,7 +6,7 @@ from odoo.tools.translate import _
 
 
 class BarcodeRule(models.Model):
-    _inherit = ['barcode.rule']
+    _inherit = 'barcode.rule'
 
     type = fields.Selection(selection_add=[
         ('weight', 'Weighted Product'),

--- a/addons/point_of_sale/models/decimal_precision.py
+++ b/addons/point_of_sale/models/decimal_precision.py
@@ -2,6 +2,7 @@ from odoo import models, api
 
 
 class DecimalPrecision(models.Model):
+    _name = 'decimal.precision'
     _inherit = ['decimal.precision', 'pos.load.mixin']
 
     @api.model

--- a/addons/point_of_sale/models/digest.py
+++ b/addons/point_of_sale/models/digest.py
@@ -6,7 +6,7 @@ from odoo.exceptions import AccessError
 
 
 class DigestDigest(models.Model):
-    _inherit = ['digest.digest']
+    _inherit = 'digest.digest'
 
     kpi_pos_total = fields.Boolean('POS Sales')
     kpi_pos_total_value = fields.Monetary(compute='_compute_kpi_pos_total_value')

--- a/addons/point_of_sale/models/ir_binary.py
+++ b/addons/point_of_sale/models/ir_binary.py
@@ -2,7 +2,7 @@ from odoo import models
 
 
 class IrBinary(models.AbstractModel):
-    _inherit = ["ir.binary"]
+    _inherit = "ir.binary"
 
     def _find_record_check_access(self, record, access_token, field):
         if record._name == "product.product" and field == "image_128":

--- a/addons/point_of_sale/models/ir_ui_view.py
+++ b/addons/point_of_sale/models/ir_ui_view.py
@@ -2,6 +2,7 @@ from odoo import models, api
 
 
 class IrUiView(models.Model):
+    _name = 'ir.ui.view'
     _inherit = ['ir.ui.view', 'pos.load.mixin']
 
     @api.model

--- a/addons/point_of_sale/models/pos_bill.py
+++ b/addons/point_of_sale/models/pos_bill.py
@@ -4,6 +4,7 @@ from odoo.osv.expression import OR
 
 
 class PosBill(models.Model):
+    _name = 'pos.bill'
     _order = "value"
     _description = "Coins/Bills"
     _inherit = ["pos.load.mixin"]

--- a/addons/point_of_sale/models/pos_bus_mixin.py
+++ b/addons/point_of_sale/models/pos_bus_mixin.py
@@ -5,6 +5,7 @@ from odoo import fields, models, api
 
 
 class PosBusMixin(models.AbstractModel):
+    _name = 'pos.bus.mixin'
     _description = "Bus Mixin"
 
     access_token = fields.Char('Security Token', copy=False)

--- a/addons/point_of_sale/models/pos_category.py
+++ b/addons/point_of_sale/models/pos_category.py
@@ -9,6 +9,7 @@ from odoo.exceptions import ValidationError, UserError
 
 
 class PosCategory(models.Model):
+    _name = 'pos.category'
     _description = "Point of Sale Category"
     _inherit = ['pos.load.mixin']
     _order = "sequence, name"

--- a/addons/point_of_sale/models/pos_config.py
+++ b/addons/point_of_sale/models/pos_config.py
@@ -13,6 +13,7 @@ from odoo.tools import convert, SQL
 
 
 class PosConfig(models.Model):
+    _name = 'pos.config'
     _inherit = ['pos.bus.mixin', 'pos.load.mixin']
     _description = 'Point of Sale Configuration'
     _check_company_auto = True

--- a/addons/point_of_sale/models/pos_load_mixin.py
+++ b/addons/point_of_sale/models/pos_load_mixin.py
@@ -3,6 +3,7 @@ from odoo import models, api
 
 
 class PosLoadMixin(models.AbstractModel):
+    _name = 'pos.load.mixin'
     _description = "PoS data loading mixin"
 
     @api.model

--- a/addons/point_of_sale/models/pos_note.py
+++ b/addons/point_of_sale/models/pos_note.py
@@ -4,6 +4,7 @@ from odoo import fields, models, api
 
 
 class PosNote(models.Model):
+    _name = 'pos.note'
     _description = 'PoS Note'
     _inherit = ['pos.load.mixin']
 

--- a/addons/point_of_sale/models/pos_order.py
+++ b/addons/point_of_sale/models/pos_order.py
@@ -20,6 +20,7 @@ _logger = logging.getLogger(__name__)
 
 
 class PosOrder(models.Model):
+    _name = 'pos.order'
     _inherit = ["portal.mixin", "pos.bus.mixin", "pos.load.mixin", "mail.thread"]
     _description = "Point of Sale Orders"
     _order = "date_order desc, name desc, id desc"
@@ -1279,6 +1280,7 @@ class PosOrder(models.Model):
 
 
 class PosOrderLine(models.Model):
+    _name = 'pos.order.line'
     _description = "Point of Sale Order Lines"
     _rec_name = "product_id"
     _inherit = ['pos.load.mixin']
@@ -1650,6 +1652,7 @@ class PosOrderLine(models.Model):
 
 
 class PosPackOperationLot(models.Model):
+    _name = 'pos.pack.operation.lot'
     _description = "Specify product lot/serial number in pos order line"
     _rec_name = "lot_name"
     _inherit = ['pos.load.mixin']
@@ -1669,6 +1672,7 @@ class PosPackOperationLot(models.Model):
 
 
 class AccountCashRounding(models.Model):
+    _name = 'account.cash.rounding'
     _inherit = ['account.cash.rounding', 'pos.load.mixin']
 
     @api.constrains('rounding', 'rounding_method', 'strategy')

--- a/addons/point_of_sale/models/pos_payment_method.py
+++ b/addons/point_of_sale/models/pos_payment_method.py
@@ -3,6 +3,7 @@ from odoo.exceptions import UserError, ValidationError
 
 
 class PosPaymentMethod(models.Model):
+    _name = 'pos.payment.method'
     _description = "Point of Sale Payment Methods"
     _order = "sequence, id"
     _inherit = ['pos.load.mixin']

--- a/addons/point_of_sale/models/pos_printer.py
+++ b/addons/point_of_sale/models/pos_printer.py
@@ -5,6 +5,7 @@ from odoo import fields, models, api
 
 
 class PosPrinter(models.Model):
+    _name = 'pos.printer'
 
     _description = 'Point of Sale Printer'
     _inherit = ['pos.load.mixin']

--- a/addons/point_of_sale/models/pos_session.py
+++ b/addons/point_of_sale/models/pos_session.py
@@ -13,6 +13,7 @@ from odoo.osv.expression import AND
 
 
 class PosSession(models.Model):
+    _name = 'pos.session'
     _order = 'id desc'
     _description = 'Point of Sale Session'
     _inherit = ['mail.thread', 'mail.activity.mixin', "pos.bus.mixin", 'pos.load.mixin']
@@ -1916,7 +1917,7 @@ class PosSession(models.Model):
 
 
 class ProcurementGroup(models.Model):
-    _inherit = ['procurement.group']
+    _inherit = 'procurement.group'
 
     @api.model
     def _run_scheduler_tasks(self, use_new_cursor=False, company_id=False):

--- a/addons/point_of_sale/models/product_attribute.py
+++ b/addons/point_of_sale/models/product_attribute.py
@@ -3,6 +3,7 @@ from odoo.osv.expression import AND
 
 
 class ProductAttribute(models.Model):
+    _name = 'product.attribute'
     _inherit = ['product.attribute', 'pos.load.mixin']
 
     @api.model
@@ -11,6 +12,7 @@ class ProductAttribute(models.Model):
 
 
 class ProductAttributeCustomValue(models.Model):
+    _name = 'product.attribute.custom.value'
     _inherit = ["product.attribute.custom.value", "pos.load.mixin"]
 
     pos_order_line_id = fields.Many2one('pos.order.line', string="PoS Order Line", ondelete='cascade')
@@ -25,6 +27,7 @@ class ProductAttributeCustomValue(models.Model):
 
 
 class ProductTemplateAttributeLine(models.Model):
+    _name = 'product.template.attribute.line'
     _inherit = ['product.template.attribute.line', 'pos.load.mixin']
 
     @api.model
@@ -38,6 +41,7 @@ class ProductTemplateAttributeLine(models.Model):
 
 
 class ProductTemplateAttributeValue(models.Model):
+    _name = 'product.template.attribute.value'
     _inherit = ['product.template.attribute.value', 'pos.load.mixin']
 
     @api.model

--- a/addons/point_of_sale/models/product_category.py
+++ b/addons/point_of_sale/models/product_category.py
@@ -3,6 +3,7 @@ from odoo import api, models
 
 
 class ProductCategory(models.Model):
+    _name = 'product.category'
     _inherit = ['product.category', 'pos.load.mixin']
 
     @api.model

--- a/addons/point_of_sale/models/product_combo.py
+++ b/addons/point_of_sale/models/product_combo.py
@@ -4,6 +4,7 @@ from odoo import api, models
 
 
 class ProductCombo(models.Model):
+    _name = 'product.combo'
     _inherit = ['product.combo', 'pos.load.mixin']
 
     @api.model

--- a/addons/point_of_sale/models/product_combo_item.py
+++ b/addons/point_of_sale/models/product_combo_item.py
@@ -4,6 +4,7 @@ from odoo import api, models
 
 
 class ProductComboItem(models.Model):
+    _name = 'product.combo.item'
     _inherit = ['product.combo.item', 'pos.load.mixin']
 
     @api.model

--- a/addons/point_of_sale/models/product_packaging.py
+++ b/addons/point_of_sale/models/product_packaging.py
@@ -3,6 +3,7 @@ from odoo.osv.expression import AND
 
 
 class ProductPackaging(models.Model):
+    _name = 'product.packaging'
     _inherit = ['product.packaging', 'pos.load.mixin']
 
     @api.model

--- a/addons/point_of_sale/models/product_pricelist.py
+++ b/addons/point_of_sale/models/product_pricelist.py
@@ -3,6 +3,7 @@ from odoo import api, fields, models
 
 
 class ProductPricelist(models.Model):
+    _name = 'product.pricelist'
     _inherit = ['product.pricelist', 'pos.load.mixin']
 
     @api.model
@@ -16,6 +17,7 @@ class ProductPricelist(models.Model):
 
 
 class ProductPricelistItem(models.Model):
+    _name = 'product.pricelist.item'
     _inherit = ['product.pricelist.item', 'pos.load.mixin']
 
     @api.model

--- a/addons/point_of_sale/models/product_product.py
+++ b/addons/point_of_sale/models/product_product.py
@@ -4,6 +4,7 @@ from odoo.exceptions import UserError
 
 
 class ProductProduct(models.Model):
+    _name = 'product.product'
     _inherit = ['product.product', 'pos.load.mixin']
 
     @api.model

--- a/addons/point_of_sale/models/product_template.py
+++ b/addons/point_of_sale/models/product_template.py
@@ -9,6 +9,7 @@ from datetime import date
 
 
 class ProductTemplate(models.Model):
+    _name = 'product.template'
     _inherit = ['product.template', 'pos.load.mixin']
 
     available_in_pos = fields.Boolean(string='Available in POS', help='Check if you want this product to appear in the Point of Sale.', default=False)

--- a/addons/point_of_sale/models/report_sale_details.py
+++ b/addons/point_of_sale/models/report_sale_details.py
@@ -10,6 +10,7 @@ from odoo.tools import SQL
 
 
 class ReportPoint_Of_SaleReport_Saledetails(models.AbstractModel):
+    _name = 'report.point_of_sale.report_saledetails'
 
     _description = 'Point of Sale Details'
 

--- a/addons/point_of_sale/models/res_company.py
+++ b/addons/point_of_sale/models/res_company.py
@@ -6,6 +6,7 @@ from odoo.osv import expression
 
 
 class ResCompany(models.Model):
+    _name = 'res.company'
     _inherit = ['res.company', 'pos.load.mixin']
 
     point_of_sale_update_stock_quantities = fields.Selection([

--- a/addons/point_of_sale/models/res_country.py
+++ b/addons/point_of_sale/models/res_country.py
@@ -2,6 +2,7 @@ from odoo import models, api
 
 
 class ResCountry(models.Model):
+    _name = 'res.country'
     _inherit = ['res.country', 'pos.load.mixin']
 
     @api.model

--- a/addons/point_of_sale/models/res_country_state.py
+++ b/addons/point_of_sale/models/res_country_state.py
@@ -2,6 +2,7 @@ from odoo import models, api
 
 
 class ResCountryState(models.Model):
+    _name = 'res.country.state'
     _inherit = ['res.country.state', 'pos.load.mixin']
 
     @api.model

--- a/addons/point_of_sale/models/res_currency.py
+++ b/addons/point_of_sale/models/res_currency.py
@@ -2,6 +2,7 @@ from odoo import models, api
 
 
 class ResCurrency(models.Model):
+    _name = 'res.currency'
     _inherit = ['res.currency', 'pos.load.mixin']
 
     @api.model

--- a/addons/point_of_sale/models/res_lang.py
+++ b/addons/point_of_sale/models/res_lang.py
@@ -2,6 +2,7 @@ from odoo import models, api
 
 
 class ResLang(models.Model):
+    _name = 'res.lang'
     _inherit = ['res.lang', 'pos.load.mixin']
 
     @api.model

--- a/addons/point_of_sale/models/res_partner.py
+++ b/addons/point_of_sale/models/res_partner.py
@@ -3,6 +3,7 @@ from odoo import api, fields, models
 
 
 class ResPartner(models.Model):
+    _name = 'res.partner'
     _inherit = ['res.partner', 'pos.load.mixin']
 
     pos_order_count = fields.Integer(

--- a/addons/point_of_sale/models/res_users.py
+++ b/addons/point_of_sale/models/res_users.py
@@ -2,7 +2,7 @@ from odoo import models, api
 
 
 class ResUsers(models.Model):
-    _inherit = ['res.users']
+    _inherit = 'res.users'
 
     @api.model
     def _load_pos_data_domain(self, data):

--- a/addons/point_of_sale/models/stock_picking.py
+++ b/addons/point_of_sale/models/stock_picking.py
@@ -10,7 +10,7 @@ from collections import defaultdict
 
 
 class StockPicking(models.Model):
-    _inherit = ['stock.picking']
+    _inherit = 'stock.picking'
 
     pos_session_id = fields.Many2one('pos.session', index=True)
     pos_order_id = fields.Many2one('pos.order', index=True)
@@ -160,6 +160,7 @@ class StockPicking(models.Model):
 
 
 class StockPickingType(models.Model):
+    _name = 'stock.picking.type'
     _inherit = ['stock.picking.type', 'pos.load.mixin']
 
     @api.depends('warehouse_id')
@@ -188,13 +189,13 @@ class StockPickingType(models.Model):
 
 
 class ProcurementGroup(models.Model):
-    _inherit = ['procurement.group']
+    _inherit = 'procurement.group'
 
     pos_order_id = fields.Many2one('pos.order', 'POS Order')
 
 
 class StockMove(models.Model):
-    _inherit = ['stock.move']
+    _inherit = 'stock.move'
 
     def _get_new_picking_values(self):
         vals = super(StockMove, self)._get_new_picking_values()

--- a/addons/point_of_sale/models/stock_rule.py
+++ b/addons/point_of_sale/models/stock_rule.py
@@ -6,7 +6,7 @@ from odoo import models
 
 
 class StockRule(models.Model):
-    _inherit = ['stock.rule']
+    _inherit = 'stock.rule'
 
     def _get_stock_move_values(self, product_id, product_qty, product_uom, location_id, name, origin, company_id, values):
         move_values = super()._get_stock_move_values(product_id, product_qty, product_uom, location_id, name, origin, company_id, values)

--- a/addons/point_of_sale/models/stock_warehouse.py
+++ b/addons/point_of_sale/models/stock_warehouse.py
@@ -4,7 +4,7 @@ from odoo import models, fields, api, _
 
 
 class StockWarehouse(models.Model):
-    _inherit = ["stock.warehouse"]
+    _inherit = "stock.warehouse"
 
     pos_type_id = fields.Many2one('stock.picking.type', string="Point of Sale Operation Type")
 

--- a/addons/point_of_sale/models/uom.py
+++ b/addons/point_of_sale/models/uom.py
@@ -2,6 +2,7 @@ from odoo import api, fields, models
 
 
 class UomCategory(models.Model):
+    _name = 'uom.category'
     _inherit = ['uom.category', 'pos.load.mixin']
 
     is_pos_groupable = fields.Boolean(string='Group Products in POS',
@@ -17,6 +18,7 @@ class UomCategory(models.Model):
 
 
 class UomUom(models.Model):
+    _name = 'uom.uom'
     _inherit = ['uom.uom', 'pos.load.mixin']
 
     is_pos_groupable = fields.Boolean(related='category_id.is_pos_groupable', readonly=False)
